<a href="https://colab.research.google.com/github/lokeshp051/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/lokeshp051/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing Refresh / Content Opportunity Scoring.
FlyRank's reviewers can't manually check every page a client owns, so the real bottleneck isn't knowing content could be improved — it's knowing which pages to look at first.
This lane produces a ranked queue instead of a single yes/no answer, which matches how a review team actually works: limited time, so rank by expected value.
It also lets me build directly on top of the starter pipeline's baseline score and label, so I can compare my own reasoning against a transparent rule from day one.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision improved: Which pages should a content reviewer prioritize this week?

Who acts on it: The SEO/content team, who have limited time to refresh, expand, or prune pages.

Cost of a wrong call: A false positive wastes a reviewer's time on a page that doesn't actually need attention, creating an opportunity cost because more important pages are delayed.
A false negative is more serious: a genuinely declining, high-traffic page may go unnoticed, causing continued loss of visibility and traffic.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
!git clone https://github.com/lokeshp051/flyrank-ml-internship.git
%cd flyrank-ml-internship

import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

total_rows = len(df)
pct_declining = round((df["trend_direction"] == "down").mean() * 100, 1)
median_impressions = df["impressions_90d"].median()

print("Total pages in starter dataset:", total_rows)
print("% currently in a declining trend:", pct_declining, "%")
print("Median impressions (90d):", median_impressions)

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 115, done.
remote: Counting objects: 100% (115/115), done.
remote: Compressing objects: 100% (70/70), done.
remote: Total 115 (delta 32), reused 102 (delta 29), pack-reused 0 (from 0)
Receiving objects: 100% (115/115), 1.84 MiB | 23.22 MiB/s, done.
Resolving deltas: 100% (32/32), done.
/content/flyrank-ml-internship/flyrank-ml-internship/flyrank-ml-internship/flyrank-ml-internship
Total pages in starter dataset: 30000
% currently in a declining trend: 54.2 %
Median impressions (90d): 731.0


"Out of 30,000 pages in the starter dataset, 54.2% are currently labeled declining, with a median of 731 impressions over 90 days — enough volume that this isn't noise, and with over half the pages showing decline, a ranked review queue is clearly worth building since reviewers can't check all 30,000 by hand."

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What my work can say:

My work can identify observed and measured patterns in this anonymized 30,000-page starter dataset, such as associations between signals like impressions, position, freshness, or word count and the declining label. All claims will use directional, decision-support language such as "this suggests," "we observed," or "pages with this characteristic were more often flagged as declining." The final output is a ranked review queue that helps a human reviewer prioritize work—it is a recommendation, not a verdict.

What my work can never say:

That I predicted or reverse-engineered Google's ranking algorithm, because this dataset only shows outcomes, not the algorithm itself.

That refreshing a page will cause it to recover, because that would require causal experiments rather than observational data. I can only identify pages that are candidates for review.

Anything involving real client names, domains, URLs, or search queries, since the dataset is fully anonymized.

That the 54.2% decline rate observed in this starter dataset represents real-world proportions. I will not generalize these findings beyond this dataset without additional evidence.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.